src/components/course/CourseContextProvider.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext, useReducer } from 'react';
+import PropTypes from 'prop-types';
+
+const initialState = {
+  course: {},
+  activeCourseRun: {},
+  userEnrollments: [],
+  userEntitlements: [],
+};
+
+export const CourseContext = createContext();
+export const CourseContextConsumer = CourseContext.Consumer;
+
+const reducer = (state, action) => {
```

Comment:
This reducer function can be used to update the values of `CourseContext`, e.g. when the learner selects a specific course run, the `set-course-run` action should be dispatched.